aiida/orm/nodes/data/array/kpoints.py

```diff
@@ -36,6 +36,36 @@ class KpointsData(ArrayData):
     set_cell_from_structure methods.
     """
 
+    def __init__(
+        self,
+        *args,
+        structuredata=None,
+        cell=None,
+        pbc=None,
+        mesh=None,
+        distance=None,
+        offset=None,
+        force_parity=False,
+        kpoints=None,
+        cartesian=False,
+        labels=None,
+        weights=None,
+        fill_values=0,
+        **kwargs
+    ):
+        """Set the properties directly in the constructor."""
+        super().__init__(*args, **kwargs)
+        if structuredata is not None:
+            self.set_cell_from_structure(structuredata)
+        if cell is not None:
+            self.set_cell(cell, pbc)
```

Comment:
Both two methods will set the cell, what if the user provides both? Same problem for the kpoint_mesh below. I think we should only allow settable properties can be set in the __init__, such as `cell`, `kpoint`, `label`.

Reply:
No worries. I just made the requested changes, ready for review now.